### DISCUSSION TITLE
The plugin deduplication feature is not working as expected.

### DISCUSSION THREAD

**Discussion by aliabdy** [elysiajs/elysia#700] [Q&A]
hi everyone
I'm reading this section ( https://elysiajs.com/essential/plugin.html#plugin-deduplication ) of the documentt and writing a sample code based on it:

```typescript
import { Elysia } from "elysia";

class Db {
  constructor(type: 'sqlite' | 'pg') {
    console.log(`create ${type} database client...`)
  }

  getPosts() {
    return []
  }
}

const database = (type: 'sqlite' | 'pg') => new Elysia({ name: 'database', seed: type })
  .decorate("db", new Db(type))

const controllerA = new Elysia()
  .use(database('sqlite'))
  .get("/posts", ({ db }) => {
      return db.getPosts()
  })

const controllerB = new Elysia()
  .use(database('sqlite'))

const app = new Elysia()
  .use(controllerA)
  .use(controllerB)
  .listen(3000);

console.log(
  `🦊 Elysia is running at ${app.server?.hostname}:${app.server?.port}`
);
``` 
```shell
create sqlite database client...
create sqlite database client...
``` 
but I see logs for creating database instances twice! what's going on? Do you know why plugin deduplication doesn't work here?

**Comment by mtt-artis** [comment]
Hey 👋 
Have you checked the scope of your middleware ?
https://elysiajs.com/essential/scope

**Comment by mtt-artis** [comment]
I think it is not a scope problem after all

Elysia skip the **registration** of Elysia instance, not it's **instantiation**.
the `use` method runs the `database` function and get a new Elysia intance with its own decorators
the duplication based on the name of each instance runs after

**Reply by aliabdy** [reply]
I think the document should clarify what is meant by registration?  Shouldn't the "decorate" be calculated only once?
